docs/reference/analysis/concepts.asciidoc

```diff
@@ -0,0 +1,11 @@
+[[analysis-concepts]]
+== Text analysis concepts
+++++
+<titleabbrev>Concepts</titleabbrev>
+++++
+
+This section explains the fundamental concepts of text analysis in {es}.
+
+* <<analyzer-anatomy>>
```

Comment:
This is a bit sparse at the moment, but I plan to move some of the index/search time analysis content here from the [top-level Analysis page](https://www.elastic.co/guide/en/elasticsearch/reference/master/analysis.html) in an upcoming PR.